cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,817 @@
+    )
+
+  # Run the partition manager algorithm.
+  execute_process(
+    COMMAND
+    ${pm_cmd}
+    RESULT_VARIABLE ret
+    )
+
+  if(NOT ${ret} EQUAL "0")
+    message(FATAL_ERROR "Partition Manager failed, aborting. Command: ${pm_cmd}")
+  endif()
+
+  # Produce header files and config file.
+  execute_process(
+    COMMAND
+    ${pm_output_cmd}
+    RESULT_VARIABLE ret
+    )
+
+  if(NOT ${ret} EQUAL "0")
+    message(FATAL_ERROR "Partition Manager output generation failed, aborting. Command: ${pm_output_cmd}")
+  endif()
+
+endfunction()
+
+
+## Check if current image is the dynamic partition in its domain.
+## I.E. it is the only partition without a statically configured size in this
+## domain. This is equivalent to the 'app' partition in the root domain.
+##
+## The dynamic partition is specified by the parent domain (i.e. the domain
+## which creates the current domain through 'create_domain_image()'.
+#if(DEFINED ${DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION
+#   AND "${IMAGE_NAME}" STREQUAL "${${DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION}"
+#)
+#  set(is_dynamic_partition_in_domain TRUE)
+#endif()
+
+get_property(PM_IMAGES GLOBAL PROPERTY PM_IMAGES)
+get_property(PM_SUBSYS_PREPROCESSED GLOBAL PROPERTY PM_SUBSYS_PREPROCESSED)
+#get_property(PM_DOMAINS GLOBAL PROPERTY PM_DOMAINS)
+
+# This file is executed once per domain.
+#
+# It will be executed if one of the following criteria is true for the
+# current image:
+# - It's a child image, and is the dynamic partition in the domain
+# - It's the root image, and a static configuration has been provided
+# - It's the root image, and PM_IMAGES is populated.
+# - It's the root image, and other domains exist.
+# - A subsys has defined a partition and CONFIG_PM_SINGLE_IMAGE is set.
+# Otherwise, return here
+#if (NOT (
+#  (IMAGE_NAME AND is_dynamic_partition_in_domain) OR
+#  (NOT IMAGE_NAME AND static_configuration) OR
+#  (NOT IMAGE_NAME AND PM_IMAGES) OR
+#  (NOT IMAGE_NAME AND PM_DOMAINS) OR
+#  (PM_SUBSYS_PREPROCESSED AND CONFIG_PM_SINGLE_IMAGE)
+#  ))
+#  return()
+#endif()
+
+# Set the dynamic partition. This is the only partition which does not
+# have a statically defined size. There is only one dynamic partition per
+# domain. For the "root domain" (ie the domain of the root image) this is
+# always "app".
+if (NOT is_dynamic_partition_in_domain)
+  set(dynamic_partition "app")  # Should this be renamed to main image name, or does it matter at all ?
+else()
+  set(dynamic_partition ${${DOMAIN}_PM_DOMAIN_DYNAMIC_PARTITION})
+  set(
+    dynamic_partition_argument
+    "--flash_primary-dynamic-partition;${dynamic_partition}"
+    )
+endif()
+
+# Image files, those should be queried from actual image.
+# Add the dynamic partition as an image partition.
+#set_property(GLOBAL PROPERTY
+#  ${dynamic_partition}_PM_HEX_FILE
+#  ${PROJECT_BINARY_DIR}/${KERNEL_HEX_NAME}
+#  )
+#
+#set_property(GLOBAL PROPERTY
+#  ${dynamic_partition}_PM_TARGET
+#  ${logical_target_for_zephyr_elf}
+#  )
```

Comment:
Commented code, can this go?

Reply:
although commented out, I prefer to keep the original intention in case we discover someone missing this info, for example during dfu package creation or signing.